crates/burn-jit/src/codegen/kernel.rs

```diff
@@ -250,7 +248,9 @@ fn execute_dynamic<R, K, E1, E2, E3>(
         handles.push(handle);
     }
 
-    let kernel: Box<dyn Kernel> = Box::new(DynamicKernel::new(kernel, workgroup));
+    let kernel = Kernel::Jit(Box::new(DynamicKernel::<K, R::Compiler>::new(
+        kernel, workgroup,
+    )));
```

Comment:
This is quite hugly, can we create a function on `DynamicKernel` that returns `Kernel` instead of `Self` ? Or we could implement `From<DynamicKernel> for Kernel` and simply call `DyanamicKernel::new(...).into())` which would be nice.